js/models/base.js

```diff
@@ -99,17 +99,19 @@ export class Base {
      * @param  {Function} on_success The callback function to call on success.
      * @param  {Function} on_error The callback function to call on error.
      */
-    $api(endpoint, data, on_success, on_error=() => {}) {
-        const parts = endpoint.split('.');
-        const namespace = parts[0];
-        const method = parts[1];
-        const operation = API[namespace][method];
-
-        if (this.$options.mask && !('X-Fields' in data)) {
-            data['X-Fields'] = mask(this.$options.mask);
-        }
+    $api(endpoint, data, on_success, on_error = () => {}) {
+        API.onReady(() => {
+            const parts = endpoint.split('.');
+            const namespace = parts[0];
+            const method = parts[1];
+            const operation = API[namespace][method];
+
+            if (this.$options.mask && !('X-Fields' in data)) {
+                data['X-Fields'] = mask(this.$options.mask);
+            }
 
-        return operation(data, on_success.bind(this), on_error.bind(this));
+            operation(data, on_success.bind(this), on_error.bind(this));
```

Comment:
I wonder if the `this` binding is required here. It probably depends on the way `on_success` has been declared, tricky :)